11-node-frameworks/classworks/Gavruk_Alexander/index.js

```diff
@@ -0,0 +1,59 @@
+const Koa = require('koa');
+const Joi = require('joi')
+const bodyParser = require('koa-bodyparser');
+const serve = require('koa-static');
+const session = require('koa-generic-session');
+const Router = require('koa-router');
+
+const router = new Router();
+
+const app = new Koa();
+
+const schema = Joi.object({
+  name: Joi.string(),
+  surname: Joi.string().required(),
+  description: Joi.string().min(3),
+  mark: Joi.number(),
+});
+
+app.keys = ['keys', 'keykeys'];
+app.use(session());
+
+app.use(serve('./static/text'));
+app.use(serve('./static/images'));
+
+app.use(bodyParser());
+
+router
+  .post('/summer', async (ctx) => {
+    const { error, value } = schema.validate(ctx.request.body);
+    if (error) {
+      ctx.body = error.details[0].message;
+    } else {
+      ctx.body = JSON.stringify(value);
+    }
+  })
+
+app.use(async (ctx, next) => {
+  updateCount.call(ctx);
```

Comment:
You need to use the counter as middleware to don't repeat this call in each other middlewares. It should work by default for each new future request (which will be added to code) without adding any code for that.